plugins/obs-outputs/librtmp/rtmp.c

```diff
@@ -1059,58 +1037,71 @@ RTMP_Connect1(RTMP *r, RTMPPacket *cp)
-
-#ifdef _WIN32
-    //COMODO security software sandbox blocks all DNS by returning "host not found"
-    h = gethostbyname("localhost");
-    if (!h && GetLastError() == WSAHOST_NOT_FOUND)
-    {
-        r->last_error_code = WSAHOST_NOT_FOUND;
-        RTMP_Log(RTMP_LOGERROR, "RTMP_Connect: Connection test failed. This error is likely caused by Comodo Internet Security running OBS in sandbox mode. Please add OBS to the Comodo automatic sandbox exclusion list, restart OBS and try again (11001).");
-        return FALSE;
+    struct happy_eyeballs_ctx* happy_ctx = NULL;
+    bool free_hostname = FALSE;
+    char *hostname = NULL;
+    int port = 0;
+    int result = FALSE;
+    int he_result = happy_eyeballs_create(&happy_ctx);
+    if (he_result != 0)
+    {
+        // did not successfully create the happy eyeballs context
+        r->last_error_code = -he_result;
+        goto fail;
     }
-#endif
-
-    memset(&service, 0, sizeof(service));
-
-    if (r->m_bindIP.addrLen)
-        addrlen_hint = r->m_bindIP.addrLen;
-
     if (r->Link.socksport)
     {
         /* Connect via SOCKS */
-        if (!add_addr_info(&service, &addrlen, &r->Link.sockshost, r->Link.socksport, addrlen_hint, &socket_error))
-        {
-            r->last_error_code = socket_error;
-            return FALSE;
-        }
+        hostname = get_hostname(&r->Link.sockshost, &free_hostname);
+        port = r->Link.socksport;
     }
     else
     {
         /* Connect directly */
-        if (!add_addr_info(&service, &addrlen, &r->Link.hostname, r->Link.port, addrlen_hint, &socket_error))
-        {
-            r->last_error_code = socket_error;
-            return FALSE;
-        }
+        hostname = get_hostname(&r->Link.hostname, &free_hostname);
+        port = r->Link.port;
+    }
+    // Set local bind address (if present)
+    happy_eyeballs_set_bind_addr(happy_ctx, r->m_bindIP.addrLen, &r->m_bindIP.addr);
+
+    // Attempt connection
+    he_result = happy_eyeballs_connect(happy_ctx, hostname, port);
+    if (he_result == EAGAIN)
+        // Connect returned with the connection process ongoing, let's wait for a few more seconds...
+        he_result = happy_eyeballs_timedwait_default(happy_ctx);
+
+    if (he_result != 0) {
+        // Error while connecting
+        int err = happy_eyeballs_get_error_code(happy_ctx);
+        if (err == E_CONNREFUSED)
+            RTMP_Log(RTMP_LOGERROR, "%s is offline. Try a different server (ECONNREFUSED).", r->Link.hostname.av_val);
+        else if (err == E_ACCES)
+            RTMP_Log(RTMP_LOGERROR, "The connection is being blocked by a firewall or other security software (EACCES).");
+        else if (err == E_TIMEDOUT)
+            RTMP_Log(RTMP_LOGERROR, "The connection timed out. Try a different server, or check that the connection is not being blocked by a firewall or other security software (ETIMEDOUT).");
+        else
+            RTMP_Log(RTMP_LOGERROR, "%s, failed to connect socket: %s (%d)",
```

Comment:
Does this produce a relevant error message if `hostname` was `NULL` or `port` was `0`? `err` should be `0` in those cases I think, so `he_result == STATUS_INVALID_ARGUMENT` should probably get special handling